### PR TITLE
Issue #2192 - SyncError app condition disappears during app reconciliation

### LINKED ISSUE #2192: SyncError app condition disappears during app reconciliation
**Describe the bug**
SyncError condition should be added to the app if auto-sync failed. The condition is set as expected but controller might delete it during next app reconciliation. 

**To Reproduce**
1. The issue is reproducible by test [TestAutoSyncSelfHealEnabled](https://github.com/argoproj/argo-cd/blob/adc6afd0118ccd7d85d5dc6e74a8b71c823a853a/test/e2e/app_autosync_test.go#L41-L41). Test periodically fails after controller removes expected condition.

**Expected behavior**
The condition should be removed only if app is synced successfully.

### PR DESCRIPTION
**Pull request by alexmt**
Closes #2192

### CODE REVIEW

File: controller/appcontroller.go [hunk 1]
```diff
@@ -753,8 +753,8 @@ func (ctrl *ApplicationController) processAppRefreshQueueItem() (processNext boo
 		}
 	}
 
-	ctrl.refreshAppConditions(app)
-	if len(app.Status.GetErrorConditions()) > 0 {
+	hasErrors := ctrl.refreshAppConditions(app)
```
Comment (alexec) [inline controller/appcontroller.go]: hmm - this is not canonical Go, I think it should return `ok` or `err`

File: test/e2e/app_autosync_test.go [hunk 2]
```diff
@@ -3,6 +3,8 @@ package e2e
 import (
 	"testing"
 
+	"github.com/stretchr/testify/assert"
```
Comment (alexec) [inline test/e2e/app_autosync_test.go]: imports not formatted

**Review by alexec — APPROVED** [review]
Conditional approval. See comments.

**Comment by alexmt** [comment]
sorry merged without reading comments. answering.

File: test/e2e/app_autosync_test.go (hunk 2 above)
Comment (alexmt) [reply test/e2e/app_autosync_test.go]: fixing in next PR

**Pull request merged**